controllers/ssp_controller.go

```diff
@@ -186,7 +198,90 @@ func cleanup(request *common.Request) error {
 	return err
 }
 
+func patchCRs(sspRequest *common.Request, kinds []string, patch []byte) error {
+	for _, kind := range kinds {
+		crs := &unstructured.UnstructuredList{}
+		crs.SetKind(kind)
+		crs.SetAPIVersion("ssp.kubevirt.io/v1")
+		err := sspRequest.Client.List(sspRequest.Context, crs)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				sspRequest.Logger.Info(fmt.Sprintf("No %s CR found", kind))
+				continue
+			} else {
+				sspRequest.Logger.Error(err, fmt.Sprintf("Error listing %s CRs: %s", kind, err))
+				return err
+			}
+		}
+		for _, item := range crs.Items {
+			err = sspRequest.Client.Patch(sspRequest.Context, &item, client.RawPatch(types.MergePatchType, patch))
+			if err != nil {
+				// Patching failed, maybe the CR just got removed? Log an error but keep going.
+				sspRequest.Logger.Error(err, fmt.Sprintf("Error pausing %s from namespace %s: %s",
+					item.GetName(), item.GetNamespace(), err))
+			}
+		}
+	}
+
+	return nil
+}
+
+func preReconcile(sspRequest *common.Request, kinds []string, operatorVersion string) error {
+	sspRequest.Instance.Status.OperatorVersion = operatorVersion
+	sspRequest.Instance.Status.TargetVersion = operatorVersion
+
+	preReconcilePatch := []byte(`{
+  "metadata":{
+    "annotations":{"kubevirt.io/operator.paused": "true"}
+  }
+}`)
+
+	return patchCRs(sspRequest, kinds, preReconcilePatch)
+}
+
+func postReconcile(sspRequest *common.Request, kinds []string, operatorVersion string) error {
```

Comment:
Please set the `ObservedVersion` in `updateStatus` function, where other status fields are updated. Then this function can be removed.